pkg/commands/config_download.go

```diff
@@ -0,0 +1,118 @@
+package commands
+
+import (
+	"errors"
+	"fmt"
+	"io/ioutil"
+	"net/url"
+	"os"
+	"syscall"
+
+	"github.com/paralus/cli/pkg/kratos"
+	"github.com/paralus/cli/pkg/log"
+	"github.com/spf13/cobra"
+	"golang.org/x/term"
+)
+
+const (
+	downloadConfigFilePathFlag = "to-file"
+	emailFlag                  = "email"
+	passwordFlag               = "password"
+)
+
+type DownloadConfigsOptions struct {
+	downloadConfigFilePath string
+	email                  string
+	password               string
+	logger                 log.Logger
+}
+
+func NewDownloadConfigOptions(logger log.Logger) *DownloadConfigsOptions {
+	o := new(DownloadConfigsOptions)
+	o.logger = logger
+	return o
+}
+
+func (o *DownloadConfigsOptions) Validate(cmd *cobra.Command, args []string) error {
+	f := cobra.ExactArgs(1)
+	if err := f(cmd, args); err != nil {
+		return err
+	}
+	_, err := url.ParseRequestURI(args[0])
+	if err != nil {
+		return err
+	}
+
+	if o.email == "" && o.password != "" || o.email != "" && o.password == "" {
+		return errors.New("please provide both email and password to login")
+	}
+
```

Comment:
Another check could be checking if provided to-file path is not a directory.
```go
	if fi, err := os.Stat(o.downloadConfigFilePath); err == nil && fi.IsDir() {
		return fmt.Errorf("%s is a directory", o.downloadConfigFilePath)
	}
```

Reply:
Added validation